vector/src/main/java/im/vector/app/features/auth/ReAuthActivity.kt

```diff
@@ -0,0 +1,225 @@
+ * limitations under the License.
+ */
+
+package im.vector.app.features.auth
+
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.os.Parcelable
+import androidx.browser.customtabs.CustomTabsCallback
+import androidx.browser.customtabs.CustomTabsClient
+import androidx.browser.customtabs.CustomTabsServiceConnection
+import androidx.browser.customtabs.CustomTabsSession
+import com.airbnb.mvrx.MvRx
+import com.airbnb.mvrx.viewModel
+import com.airbnb.mvrx.withState
+import im.vector.app.R
+import im.vector.app.core.di.ScreenComponent
+import im.vector.app.core.extensions.addFragment
+import im.vector.app.core.platform.SimpleFragmentActivity
+import im.vector.app.core.utils.openUrlInChromeCustomTab
+import kotlinx.parcelize.Parcelize
+import org.matrix.android.sdk.api.auth.AuthenticationService
+import org.matrix.android.sdk.api.auth.data.LoginFlowTypes
+import org.matrix.android.sdk.internal.auth.registration.RegistrationFlowResponse
+import org.matrix.android.sdk.internal.auth.registration.nextUncompletedStage
+import timber.log.Timber
+import javax.inject.Inject
+
+class ReAuthActivity : SimpleFragmentActivity(), ReAuthViewModel.Factory {
+
+    @Parcelize
+    data class Args(
+            val flowType: String?,
+            val title: String?,
+            val session: String?,
+            val lastErrorCode: String?,
+            val resultKeyStoreAlias: String
+    ) : Parcelable
+
+    // For sso
+    private var customTabsServiceConnection: CustomTabsServiceConnection? = null
+    private var customTabsClient: CustomTabsClient? = null
+    private var customTabsSession: CustomTabsSession? = null
+
+    @Inject lateinit var authenticationService: AuthenticationService
+    @Inject lateinit var reAuthViewModelFactory: ReAuthViewModel.Factory
+
+    override fun create(initialState: ReAuthState) = reAuthViewModelFactory.create(initialState)
+
+    override fun injectWith(injector: ScreenComponent) {
+        super.injectWith(injector)
+        injector.inject(this)
+    }
+
+    private val sharedViewModel: ReAuthViewModel by viewModel()
+
+    // override fun getTitleRes() = R.string.re_authentication_activity_title
+
+    override fun initUiAndData() {
+        super.initUiAndData()
+
+        val title = intent.extras?.getString(EXTRA_REASON_TITLE) ?: getString(R.string.re_authentication_activity_title)
+        supportActionBar?.setTitle(title) ?: run { setTitle(title) }
+
+//        val authArgs = intent.getParcelableExtra<Args>(MvRx.KEY_ARG)
+
+        // For the sso flow we can for now only rely on the fallback flow, that handles all
+        // the UI, due to the sandbox nature of CCT (chrome custom tab) we cannot get much information
+        // on how the process did go :/
+        // so we assume that after the user close the tab we return success and let caller retry the UIA flow :/
+
+        addFragment(
```

Comment:
Test if this is the first creation of the Activity? I think rotating the screen will add another Fragment

Reply:
+1